resourcematch-api/main.py

```diff
@@ -29,6 +56,45 @@ class MatchResponse(BaseModel):
     urls: List[AnyHttpUrl]
 
 
+async def process_search_queries(search_queries, books):
+    async with aiohttp.ClientSession() as session:
+        first_book_slug = books[0]
+        book_data = BOOKS_BY_SLUG[first_book_slug]
+        previously_used_queries = {}
+        aggregate_search_results = []
+
+        for query in search_queries:
+            if query in previously_used_queries:
+                continue
+            previously_used_queries[query] = True
+            url = (f'https://openstax.org/open-search/'
+                   f'api/v0/search?q={query}'
+                   f'&books={book_data['code_version']}/{book_data['uuid']}'
+                   f'@{book_data['book_version']}'
+                   f'&index_strategy=i1&search_strategy=s1')
+            try:
+                async with session.get(url) as resp:
+                    resp.raise_for_status()
+                    if resp.status == 200:
```

Comment:
Hmmm. I interpreted `raise_for_status()` as only being raised for status codes 400 or higher per these [docs](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.raise_for_status). It explicitly states do nothing for success responses, so I thought the additional check may be useful. Is that your understanding as well?

Reply:
That's a correct understanding, but what other non-`4xx/`5xx` would we expect that we would want to treat as an error?

Reply:
Ahhh, I see what you are saying. I guess for connection errors and timeout errors we wouldn't even get to that code path because the `session.get()` would fail?

Reply:
Ah -- yep, that's right.

Reply:
Thanks for the clarification - removed the `200` check.